CMakeLists.txt

```diff
@@ -2,17 +2,16 @@ if(CONFIG_HAS_MCUX OR CONFIG_HAS_IMX_HAL OR CONFIG_HAS_S32_HAL)
   zephyr_library()
 endif()
 
-add_subdirectory_ifdef(
-  CONFIG_HAS_MCUX
-  mcux
-  )
+if(CONFIG_HAS_S32_HAL)
+  add_subdirectory(s32)
```

Comment:
why did you have to structure things this way?

Reply:
Right. When using MCUX with S32, MCUX sources are built from the ./s32/CMakeLists.txt. We don't want to include also the ./mcux/CMakeLists.txt. This allow to build S32 with `CONFIG_HAS_S32_HAL` and `CONFIG_HAS_MCUX`.